include/aca_zeta_state_handler.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 The Alcor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ACA_ZETA_STATE_HANDLER_H
+#define ACA_ZETA_STATE_HANDLER_H
+
+#include "aca_zeta_programming.h"
+#include "goalstateprovisioner.grpc.pb.h"
+
+namespace aca_zeta_state_handler
+{
+class Aca_Zeta_State_Handler {
+  public:
+  static Aca_Zeta_State_Handler &get_instance();
+
+  //process ONE Zeta state int
+  int update_zeta_state_workitem(const alcor::schema::VpcState current_VpcState);
```

Comment:
since Zeta gateway info doesn't came as a standalone state from the goal state message, therefore, I don't think we need to create another class to handle it.

The right place to do it would be during new port create:
. it will look up the corresponding vpc state
. determine if it is a zeta support port by looking into the vpc state
. if it is zeta support:
.. add the group bucket rule if this is the first port in the VPC
.. add the OAM punt rule also if this is the first port in the VPC

delete the rules if the last port in the VPC has been deleted

Note that I am doing changes to support port deletion, therefore, we will need to co-ordinate our changes to avoid conflict.